druid-shell/src/platform/web/window.rs

```diff
@@ -230,16 +221,19 @@ fn setup_resize_callback(ws: &Rc<WindowState>) {
     let state = ws.clone();
     register_window_event_listener(ws, "resize", move |_: web_sys::UiEvent| {
         let (css_width, css_height, dpr) = state.get_window_size_and_dpr();
-        let physical_width = (dpr * css_width) as u32;
-        let physical_height = (dpr * css_height) as u32;
-        state.dpr.replace(dpr);
-        state.canvas.set_width(physical_width);
-        state.canvas.set_height(physical_height);
-        let _ = state.context.scale(dpr, dpr);
-        state
-            .handler
-            .borrow_mut()
-            .size(physical_width, physical_height);
+        let size_dp = state.scale.try_borrow_mut().map_or(None, |mut scale| {
+            *scale = Scale::from_scale(dpr, dpr);
+            let size_px = scale.set_size_dp(Size::new(css_width, css_height));
+            state.canvas.set_width(size_px.width as u32);
+            state.canvas.set_height(size_px.height as u32);
+            let _ = state.context.scale(scale.scale_x(), scale.scale_y());
+            Some(scale.size_dp())
+        });
```

Comment:
does it make sense to just move this to a method on the state type? `try_update_scale` or something? Not sure, just seems like maybe.

Reply:
Sure, why not. I moved it to its own method.